### PR TITLE
TE5-312_Swift_Exercise_Completed

### PR DESCRIPTION
**Pull request by buddhdev-jay**

### CODE REVIEW

File: IOS_DEMO/IOS_DEMO/SecondViewController.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import UIKit
+
+protocol TextDelegate {
+    func sendText(text : String)
+}
+class SecondViewController: UIViewController {
+    
+    @IBOutlet weak var txtFlied: UITextField!
+    var txtDelegate : TextDelegate?
```
Comment (purvesh-d) [inline IOS_DEMO/IOS_DEMO/SecondViewController.swift:16]: delegate is strong reference so use weak var

File: IOS_DEMO/IOS_DEMO/SecondViewController.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+import UIKit
+
+protocol TextDelegate {
+    func sendText(text : String)
+}
+class SecondViewController: UIViewController {
+    
+    @IBOutlet weak var txtFlied: UITextField!
```
Comment (purvesh-d) [inline IOS_DEMO/IOS_DEMO/SecondViewController.swift:15]: PUT //MARK:-  Outlets

File: IOS_DEMO/IOS_DEMO/ViewController.swift [hunk 3]
```diff
@@ -9,11 +9,23 @@ import UIKit
 
 class ViewController: UIViewController {
 
+    @IBOutlet weak var btn1: UIButton!
```
Comment (purvesh-d) [inline IOS_DEMO/IOS_DEMO/ViewController.swift:12]: Add outlets mark

File: Swift_Advance_Exercise.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,713 @@
...
+//8.Create a swift class example which has a parent class for vehicles, and child classes for two different vehicles e.g. (bike & car). You need to use common properties and method in the parent class and inside child class, there will be some different property which is not common.
+class vehicles {
+    var seats: Int
+    var name: String
+    init(seats:  Int, name : String) {
+        self.seats = seats
+        self.name = name
+    }
+}
+class Bike : vehicles {
```
Comment (purvesh-d) [inline Swift_Advance_Exercise.playground/Contents.swift:131]: `class Bike: vehicles {` it should be like this. Before semicolon space is not required. Check this in whole file

File: Swift_Advance_Exercise.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,713 @@
...
+    }
+    func doPower(power : Double) -> Double {
+        return pow(value, power)
+    }
+}
+let instance = Power(value: 5)
+let result = instance.doPower(power: 3)
+print(result)
+//8.Create a swift class example which has a parent class for vehicles, and child classes for two different vehicles e.g. (bike & car). You need to use common properties and method in the parent class and inside child class, there will be some different property which is not common.
+class vehicles {
```
Comment (purvesh-d) [inline Swift_Advance_Exercise.playground/Contents.swift:123]: Class name start with capital case

File: Swift_Advance_Exercise.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,713 @@
...
+}
+var examplestruct1 = Examplestruct()
+examplestruct1.a = "Hello"
+var examplestruct2 = example1
+examplestruct2.a = "Hi"
+print(examplestruct1.a)
+print(examplestruct2.a)
+//Enumerations
+//1.Write swift program using enumerations to check week day from given number. Consider Monday as 1 and Sunday as 7. We can be able to get day name as string in short form (sun, mon,...) and same full name (Sunday, Monday...) by given number of the day.
+enum weekdays {
```
Comment (purvesh-d) [inline Swift_Advance_Exercise.playground/Contents.swift:265]: Enum name start with capital case `enum Weekdays {`

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: IOS_DEMO/IOS_DEMO/SecondViewController.swift (hunk 2 above)
Comment (purvesh-d) [reply IOS_DEMO/IOS_DEMO/SecondViewController.swift:16]: Proper indentation formate

File: IOS_DEMO/IOS_DEMO/ViewController.swift (hunk 3 above)
Comment (purvesh-d) [reply IOS_DEMO/IOS_DEMO/ViewController.swift:13]: Proper indentation formate

File: Swift_Advance.playground/Contents.swift [hunk 7]
```diff
@@ -1,3 +1,266 @@
...
+print("student result is \(set2.average.m1) \(set2.average.m2) \(set2.result.mark1) \(set2.result.mark2)")
+//Convenience Initializers
+class mainClass {
+    var no1: Int
+    init(no1 : Int) {
+        self.no1 = no1
+    }
+}
+class subClass : mainClass {
+    var no2 : Int
```
Comment (purvesh-d) [inline Swift_Advance.playground/Contents.swift:135]: Space is not required after semicolon. `var no2: Int`

Comment (purvesh-d) [reply Swift_Advance.playground/Contents.swift:135]: Also var name should be proper

File: Swift_Advance_Exercise.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,713 @@
+import UIKit
+
+//Class
+//1.Create a swift program to demonstrate swift class. Which has two variables inside class body. Access (Set its values and get its value.) these variables by creating one instance of class.
+class Demo {
+    var name : String
```
Comment (purvesh-d) [inline Swift_Advance_Exercise.playground/Contents.swift:6]: same ` var name: String` Check in whole file

File: Swift_Advance_Exercise.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,713 @@
...
+        self.length = length
+        self.height = height
+    }
+    func calculate()-> Double {
+        area = Double(length * height)
+        return area
+    }
+}
+class Circle: Shape {
+    var radius : Double
```
Comment (purvesh-d) [inline Swift_Advance_Exercise.playground/Contents.swift:79]: same comment

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: Swift_Advance_Exercise.playground/Contents.swift (hunk 9 above)
Comment (buddhdev-jay) [reply Swift_Advance_Exercise.playground/Contents.swift:79]: Okk i will resolve it

**Pull request merged**